### PR TITLE
P/L testcases

### PR DESCRIPTION
**Pull request by robert-e-davidson3**
Testcases from https://docs.google.com/spreadsheets/d/1ZOw3js_EKqyJiaemouVtFdksRKPjP_o681T6alii7Gk/edit?ts=5e8cdc13#gid=0

### CODE REVIEW

File: packages/augur-tools/src/libs/TestEthersProvider.ts [hunk 1]
```diff
@@ -25,9 +25,10 @@ export class TestEthersProvider extends EthersProvider {
     return makeProvider(seed, this.accounts);
   };
 
-  getConfig(): SDKConfiguration {
+  getConfig(overwrites?: RecursivePartial<SDKConfiguration>): SDKConfiguration {
```
Comment (pgebheim) [inline packages/augur-tools/src/libs/TestEthersProvider.ts:28]: Oh that's a nice API!

File: packages/augur-test/src/tests/state/getter/users/get-profit-loss-and-summary.test.ts [hunk 2]
```diff
@@ -36,147 +40,502 @@ describe('State API :: Users :: ', () => {
...
+  await user.doInitialReport(market, outcomeList);
+}
+
+async function claimProceeds(market: Market, users: TestContractAPI[]) {
+  for (const user of users) {
+    await user.claimTradingProceeds(market);
+  }
+}
+
+function repeat<T>(thing: T, times: number): T[] {
```
Comment (pgebheim) [inline packages/augur-test/src/tests/state/getter/users/get-profit-loss-and-summary.test.ts:535]: You should put this in augur-utils :-)

**Review by pgebheim — APPROVED** [review]
One real comment to move the repeat<T> into augur-utils (I think we should start building this out)

File: packages/augur-test/src/tests/state/getter/users/get-profit-loss-and-summary.test.ts (hunk 2 above)
Comment (robert-e-davidson3) [reply packages/augur-test/src/tests/state/getter/users/get-profit-loss-and-summary.test.ts:535]: good idea! https://github.com/AugurProject/augur/pull/7363

**Pull request merged**